src/plugins/embeddable/public/lib/containers/container.ts

```diff
@@ -58,20 +58,7 @@ export abstract class Container<
     super(input, output, parent);
     this.getFactory = getFactory; // Currently required for using in storybook due to https://github.com/storybookjs/storybook/issues/13834
 
-    // initialize all children on the first input change. Delayed so it is run after the constructor is finished.
-    this.getInput$()
-      .pipe(delay(0), take(1))
-      .subscribe(() => {
-        this.initializeChildEmbeddables(input, settings);
-      });
-
-    // on all subsequent input changes, diff and update children on changes.
-    this.subscription = this.getInput$()
-      // At each update event, get both the previous and current state.
-      .pipe(pairwise())
-      .subscribe(([{ panels: prevPanels }, { panels: currentPanels }]) => {
-        this.maybeUpdateChildren(currentPanels, prevPanels);
-      });
```

Comment:
If I am understanding the problem correctly, we can reduce the number of changes needed by using RxJS operators to describe the async flow we want. This is what I came up with using `mergeMap`, `combineLatestWith` and putting the two streams together - looks to be working based on latest tests.

```ts
    // if there is no special initialization logic, we can immediately start updating children on input updates.
    const awaitingInitialize = Boolean(
      settings?.initializeSequentially || settings?.childIdInitializeOrder
    );

    // initialize all children on the first input change. Delayed so it is run after the constructor is finished.
    const init$ = this.getInput$().pipe(
      take(1),
      mergeMap(async () => {
        const initPromise = this.initializeChildEmbeddables(input, settings);
        if (awaitingInitialize) await initPromise;
      })
    );

    // on all subsequent input changes, diff and update children on changes.
    const update$ = this.getInput$()
      // At each update event, get both the previous and current state.
      .pipe(pairwise());

    this.subscription = init$
      .pipe(combineLatestWith(update$))
      .subscribe(([_, [{ panels: prevPanels }, { panels: currentPanels }]]) => {
        this.maybeUpdateChildren(currentPanels, prevPanels);
      });
```